### ISSUE TITLE
Actor Rigging Issues

### ISSUE THREAD

**Issue by ak2yny** [ak2yny/Official-Character-Pack-v2-for-Ultimate-Alliance#28]
Reports by UltraMegaMagnus

Playable:
- [ ] Many fixes needed for models with bad normals.
- [ ] Many fixes needed for models with bad fingers (too spread out). Can be fixed with actorConverter.
- [ ] MrK plans to fix Thor (Ultimate, feet are not smooth), Captain America (Ultimate, r thigh).
- [ ] All characters with cape could use a better rigging to avoid clipping, but difficult to deal with.
- [ ] Nick Fury: All of his skins have finger issues akin to Elektra (when fist are closed), I forgot how I fixed this for Elektra (if it was a rigging or fingers being too spread out).
- [ ] Nick Fury: Clipping between left arm & left strap. Probably can't be fixed without causing other oddity.
- [ ] Doctor Strange - Royal Seer: There is still some issue with the cape.
- [ ] Spider-Woman - Skin 3: Wings looking a bit odd. Looks difficult to rig.
- [ ] Thor: Some skin don't have the hammer perfectly positioned to avoid clipping during blocking animation. Impossible to fix for Ultimate Thor, because hammer is too short. Unless we make the hammer longer, but that will denaturate the hammer design?
- [ ] Moon Knight: Bad cape rigging everywhere, when running. UMM finds this too difficult to fix.

NPC:
- [ ] Mephisto: Small deformation on left upper arm. I forgot where exactly, must check playthrough, not very noticeable.
- [ ] Office Worker: as reported by Enchlore. forgot what it was about, maybe already fixed.
- [ ] Neutron: Wasted meshsmooth.

**Comment by UltraMegaMagnus** [comment]
With nikita's tools, I think it will be better to convert again the Gold skins, and maybe the PSP to have 100% accurate skeletons. I thought of using directly Gold files, but maybe they can causes crash during alt-tab as observed with Venom's tongue actor.

**Comment by ak2yny** [comment]
I remember that you fixed some rigging or the OG skins. Or is my memory playing tricks with me?

I believe that IGB files that ran through Alchemy 5 are fine in MUA. Venom's animations never went trough Alchemy 5.

**Comment by ak2yny** [comment]
Can anyone test if the crash is fixed with this file: [venom_tongue.zip](https://github.com/ak2yny/Official-Character-Pack-v2-for-Ultimate-Alliance/files/10856104/venom_tongue.zip)?

**Comment by JordanLeich** [comment]
Tested, no crashes with this https://github.com/ak2yny/Official-Character-Pack-v2-for-Ultimate-Alliance/commit/98b8181e8268f5cacc485dd09b4ca78bd75f5ead

**Comment by UltraMegaMagnus** [comment]
> I remember that you fixed some rigging or the OG skins. Or is my memory playing tricks with me?
> 
> I believe that IGB files that ran through Alchemy 5 are fine in MUA. Venom's animations never went trough Alchemy 5.

Yep, mainly to re-apply bump textures that broke lightning when using Gold files (at the time it was the only way I saw to fix that), but if skeleton is not matching perfectly MUA skeleton, we can see bones difference sometimes in game, clavicules enlarging, or moving.

**Comment by ak2yny** [comment]
I worked with MUA animations, and I found that the skeleton in animations can be changed through the animation mixer. However, the character animations (e.g. 176_venom.igb) match the original character skin skeleton and should probably be left alone.
If you use the animation mixer without any manual changes, you'll use the default skeleton (from fightstyle_default) which will change the set to use the skeleton that the common animations use. The original animations with a skeleton matching the original skin, sometimes have bad transitions to common animations, since they are made for different skeletons. I haven't seen major issues with original characters.

**Comment by ak2yny** [comment]
Jayglass made a new rig for Doom Classic, which looks nice but hasn't properly tested, yet.
https://discord.com/channels/449510825385000960/601372940440829954/1140931366440153208
